pkg/formatter/csv_test.go

```diff
@@ -131,3 +187,40 @@ func TestMakeNodesReportCSV_WithRecords(t *testing.T) {
 		assert.Equal(t, "", lines[4])
 	}
 }
+
+func TestMakeNodesReportCSV_WithMultipleNodesAndCookbooks(t *testing.T) {
+	nodesReport := []*reporting.NodeReportItem{
+		&reporting.NodeReportItem{Name: "node1", ChefVersion: "12.22", OS: "windows", OSVersion: "10.1",
+			CookbookVersions: []reporting.CookbookVersion{
+				reporting.CookbookVersion{Name: "mycookbook1", Version: "1.0"},
+				reporting.CookbookVersion{Name: "mycookbook1", Version: "2.0"},
+			},
+		},
+		&reporting.NodeReportItem{Name: "node1", ChefVersion: "13.10", OS: "windows", OSVersion: "10.1",
+			CookbookVersions: []reporting.CookbookVersion{
+				reporting.CookbookVersion{Name: "mycookbook1", Version: "1.0"},
+				reporting.CookbookVersion{Name: "mycookbook1", Version: "2.0"},
+			},
+		},
+		&reporting.NodeReportItem{Name: "node1", ChefVersion: "15.2", OS: "windows", OSVersion: "10.1",
+			CookbookVersions: []reporting.CookbookVersion{},
+		},
+		&reporting.NodeReportItem{Name: "node2", ChefVersion: "13.11", OS: "", OSVersion: "",
+			CookbookVersions: []reporting.CookbookVersion{
+				reporting.CookbookVersion{Name: "mycookbook", Version: "1.0"},
+				reporting.CookbookVersion{Name: "test", Version: "9.9"},
+			},
+		},
+		&reporting.NodeReportItem{Name: "node3", ChefVersion: "15.00", OS: "ubuntu", OSVersion: "16.04",
+			CookbookVersions: nil},
+	}
+
+	lines := strings.Split(subject.MakeNodesReportCSV(nodesReport).Report, "\n")
+	if assert.Equal(t, 7, len(lines)) {
+		assert.Equal(t, "Node Name,Chef Version,Operating System,Cookbooks", lines[0])
+		assert.Equal(t, "node1,12.22,windows v10.1,mycookbook1(1.0) mycookbook1(2.0)", lines[1])
+		assert.Equal(t, "node1,13.10,windows v10.1,mycookbook1(1.0) mycookbook1(2.0)", lines[2])
+		assert.Equal(t, "node3,15.00,ubuntu v16.04,None", lines[3])
```

Comment:
I think you need to change this assertion to be a `Contains` since the order of the entries are not yet sorted. (I created cards to do so but we haven't done it https://github.com/chef/chef-analyze/issues/98 https://github.com/chef/chef-analyze/issues/97 )